tools/dv/samples/kmdf/pool_leak/src/driver.rs

```diff
@@ -0,0 +1,200 @@
+            const { assert!(WDF_DRIVER_CONFIG_SIZE <= ULONG::MAX as usize) }
+
+            wdf_driver_config_size = WDF_DRIVER_CONFIG_SIZE as ULONG;
+        }
+
+        WDF_DRIVER_CONFIG {
+            Size: wdf_driver_config_size,
+            EvtDriverDeviceAdd: Some(evt_driver_device_add),
+            EvtDriverUnload: Some(evt_driver_unload),
+            ..WDF_DRIVER_CONFIG::default()
+        }
+    };
+
+    let driver_handle_output = WDF_NO_HANDLE.cast::<WDFDRIVER>();
+
+    let nt_status = unsafe {
+        macros::call_unsafe_wdf_function_binding!(
+            WdfDriverCreate,
+            driver as PDRIVER_OBJECT,
+            registry_path,
+            WDF_NO_OBJECT_ATTRIBUTES,
+            &mut driver_config,
+            driver_handle_output,
+        )
+    };
+
+    if !nt_success(nt_status) {
+        println!("Error: WdfDriverCreate failed {nt_status:#010X}");
+        return nt_status;
+    }
+
+    nt_status
+}
+
+/// `EvtDeviceAdd` is called by the framework in response to `AddDevice`
+/// call from the `PnP` manager. We create and initialize a device object to
+/// represent a new instance of the device.
+///
+/// # Arguments:
+///
+/// * `_driver` - Handle to a framework driver object created in `DriverEntry`
+/// * `device_init` - Pointer to a framework-allocated `WDFDEVICE_INIT`
+///   structure.
+///
+/// # Return value:
+///
+///   * `NTSTATUS`
+#[link_section = "PAGE"]
+extern "C" fn evt_driver_device_add(
+    _driver: WDFDRIVER,
+    mut device_init: *mut WDFDEVICE_INIT,
+) -> NTSTATUS {
+    paged_code!();
+
+    println!("Enter: EchoEvtDeviceAdd");
+
+    let mut attributes = WDF_OBJECT_ATTRIBUTES {
+        Size: core::mem::size_of::<WDF_OBJECT_ATTRIBUTES>() as ULONG,
+        ExecutionLevel: _WDF_EXECUTION_LEVEL::WdfExecutionLevelInheritFromParent,
+        SynchronizationScope: _WDF_SYNCHRONIZATION_SCOPE::WdfSynchronizationScopeInheritFromParent,
+        ContextTypeInfo: wdf_get_context_type_info!(DeviceContext),
+        ..WDF_OBJECT_ATTRIBUTES::default()
+    };
+
+    let mut device = WDF_NO_HANDLE as WDFDEVICE;
+    let mut nt_status = unsafe {
+        macros::call_unsafe_wdf_function_binding!(
+            WdfDeviceCreate,
+            &mut device_init,
+            &mut attributes,
+            &mut device,
+        )
+    };
+
+    if !nt_success(nt_status) {
+        println!("Error: WdfDeviceCreate failed {nt_status:#010X}");
+        return nt_status;
+    }
+
+    // Get the device context and initialize it. WdfObjectGet_DEVICE_CONTEXT is an
+    // inline function generated by WDF_DECLARE_CONTEXT_TYPE macro in the
+    // device.h header file. This function will do the type checking and return
+    // the device context. If you pass a wrong object  handle
+    // it will return NULL and assert if run under framework verifier mode.
+    let device_context = unsafe { wdf_object_get_device_context(device as WDFOBJECT) };
+    unsafe {
+        let length: usize = 64;
```

Comment:
This 64 seems like a magic number.  Where did it come from?

Reply:
I arbitrarily chose 64 bytes as the number of bytes to be allocated by ExAllocatePool2

Reply:
OK.  I might make a comment at this line that this is the memory we intend to leak, and then update the comment in evt_driver_unload to note that this line is where the leaked allocation happened, just to make it easier for people to read.